### PR TITLE
Don't let nearby monsters automatically attack hero if he moved to this tile using Stone Lith, Whirlpool or Dimension Door

### LINKED ISSUE #717: Hero shouldn't be automatically attacked by creatures placed near Stone Liths
Currently If a hero enters Stone Liths, which exit is located near creature, battle automatically occurs. 
Hero should have an ability to run away using "Space" if creatures are too strong near the Stone Liths exit.
Teleport
Teleport2
Note, that battle still should occur, when hero used "Dimension Door" spell too close to creature.

### LINKED ISSUE #1183: Sounds of world map structures continue playing till player closes dialog window after the "Dimension Door" spell
Player, standing near any world map structure, that has its own sound (ex:mines), uses "Dimension Door" spell to attack any creature. After the hero appears near creature, we scan see a pop up window (ex: creatures joining us) the sound of mine is still playing. 
This sound should be stopped before the moment we see pop up window.

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #717
fix #1183 (sort of, hero should not attack creatures when moving using the Dimension Door)

### CODE REVIEW

**Comment by felix642** [comment]
I'm not quite sure to understand the purpose here. I might be missing something. 
You added a bool to check if we have to execute the function or not? But can't you simply not call this function when you don't want monsters to attack or does calling the function with the parameter equal to false actually does something I don't see?

**Comment by oleg-derevenetz** [comment]
Hi @felix642 This bool is checked only for the part of the function (related to monsters), but there is more logic in it. For example, we still should reset `VISIONS` mode for hero after movement (even if such movement was performed using teleportation), what hasn't been done before for Dimension Door case (which was previously implemented as you suggest - by "just not calling this function) :) There is also may be some logic (in future) in `AI::HeroesActionNewPosition()`, which will not be called as well if you will not call this function.

**Review by felix642 — APPROVED** [review]

File: src/fheroes2/kingdom/kingdom.cpp [hunk 1]
```diff
@@ -609,7 +609,7 @@ void Kingdom::HeroesActionNewPosition() const
 {
     // Heroes::ActionNewPosition: can remove elements from heroes vector.
     KingdomHeroes heroes2( heroes );
-    std::for_each( heroes2.begin(), heroes2.end(), []( Heroes * hero ) { hero->ActionNewPosition(); } );
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom.cpp:612]: Shall we remove this method? It's not in use anywhere.

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left just one comment. Could you please take a look?

File: src/fheroes2/kingdom/kingdom.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/kingdom/kingdom.cpp:612]: Oh, I didn't noticed that it is not used anymore. I'll remove it.

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you so much for this fix!

**Pull request merged**